src/vision/detector.cpp

```diff
@@ -1,19 +1,128 @@
 #include "detector.h"
 
 std::vector<DetectedCard> Detector::detectCards(Images images) {
-    return std::vector<DetectedCard>();
+
+	std::vector<std::vector<cv::Point>> contours;
+	cv::findContours(images.canny, contours, cv::RETR_TREE, cv::CHAIN_APPROX_SIMPLE);
+
+	// For each detected contour, create a rectangle object in its location and add it to the
+	// detected card vector to be returned to renderer. 
+
+	for (auto contour : contours) {
+		std::vector<cv::Point> contoursPoly;
+		cv::approxPolyDP(contour, contoursPoly, 3, true);
+
+		cv::Rect card = cv::boundingRect(contoursPoly);
+
+		if (isCardValid(card, detectedCards)) {
+			detectedCards.push_back(addCardData(images, card));
+		}
+	}
+	return std::vector<DetectedCard>();
```

Comment:
This function is alway going to return an empty vector. It needs to return detectedCards instead